components/views/profile.js

```diff
@@ -24,27 +32,53 @@ const ProfileScreen = () => {
       setPhone(user.phone);
       setEmail(user.email);
       setLocation(user.location);
-      getData();
+      onRefresh();
     });
-  }, [myContext]);
+  }, [myContext.userID]);
+
+  const onRefresh = () => {
+    setIsFetching(true);
+    getData();
+    setIsFetching(false);
+  };
 
-  const getData = () => {
+  const getData = async () => {
     let temp;
-    Database.getItem("data").then((data) => {
+    await Database.getItem("data").then((data) => {
```

Comment:
There is a possibility although small that the data will not update before after the method is called. Only a problem if you update a view through code and not refresh manually.

Reply:
How do you want it handled?

Reply:
Medium rare with ocean salt on half

Reply:
Just leave it. Took it through a break test, it only acts up when it stutters a lot. That's is common for promises apparently